Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1235,6 +1238,11 @@ internal extension Purchases {
 
 private extension Purchases {
 
+    func handleCustomerInfoChanged(_ customerInfo: CustomerInfo) {
+        self.trialOrIntroPriceEligibilityChecker.clearCache()
```

Comment:
@aboedo per our discussion, this ordering now ensures there are no race conditions.